src/mapping/impl/BasisFunctions.hpp

```diff
@@ -177,31 +173,29 @@ class Gaussian : public CompactSupportBase {
 class CompactThinPlateSplinesC2 : public CompactSupportBase {
 public:
   explicit CompactThinPlateSplinesC2(double supportRadius)
-      : _r(supportRadius)
   {
-    PRECICE_CHECK(math::greater(_r, 0.0),
+    PRECICE_CHECK(math::greater(supportRadius, 0.0),
                   "Support radius for radial-basis-function compact thin-plate-splines c2 has to be larger than zero. Please update the \"support-radius\" attribute.");
+    _r_inv = 1. / supportRadius;
   }
 
   double getSupportRadius() const
   {
-    return _r;
+    return 1. / _r_inv;
   }
 
-  double evaluate(double radius) const
+  inline double evaluate(double radius) const
   {
-    if (radius >= _r)
+    double const p = radius * _r_inv;
+    if (p >= 1)
       return 0.0;
-    double const p = radius / _r;
-    using std::log;
-    using std::pow;
-    return 1.0 - 30.0 * pow(p, 2.0) - 10.0 * pow(p, 3.0) + 45.0 * pow(p, 4.0) - 6.0 * pow(p, 5.0) - 60.0 * log(pow(p, pow(p, 3.0)));
+    return 1.0 - 30.0 * math::pow_int<2>(p) - 10.0 * math::pow_int<3>(p) + 45.0 * math::pow_int<4>(p) - 6.0 * math::pow_int<5>(p) - math::pow_int<3>(p) * 60.0 * std::log(std::max(p, 1e-15));
```

Comment:
Instead of going for this overly complicated formula in the end log(p^(p^3)) I selected now a fixed cutoff for the log (1e-14). The overall product will be zero anyway, as well multiply it by (1e-14)^3.